src/main/java/tw/joi/energy/domain/PricePlan.java

```diff
@@ -1,64 +1,55 @@
 package tw.joi.energy.domain;
 
-import java.io.*;
 import java.math.BigDecimal;
-import java.text.*;
 import java.time.DayOfWeek;
-import java.time.LocalDateTime;
-import java.util.List;
+import java.time.ZonedDateTime;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
 
 public class PricePlan {
 
-    private String energySupplier;
-    private String planName;
+    private final String energySupplier;
+    private final String planName;
     private final BigDecimal unitRate; // unit price per kWh
-    private final List<PeakTimeMultiplier> peakTimeMultipliers;
+    private final Map<DayOfWeek, BigDecimal> peakTimeMultipliers;
+
+    public PricePlan(String planName, String energySupplier, BigDecimal unitRate) {
+        this.planName = planName;
+        this.energySupplier = energySupplier;
+        this.unitRate = unitRate;
+        this.peakTimeMultipliers = Collections.emptyMap();
+    }
 
     public PricePlan(
-            String planName, String energySupplier, BigDecimal unitRate, List<PeakTimeMultiplier> peakTimeMultipliers) {
+            String planName,
+            String energySupplier,
+            BigDecimal unitRate,
+            Map<DayOfWeek, BigDecimal> peakTimeMultipliers) {
         this.planName = planName;
         this.energySupplier = energySupplier;
         this.unitRate = unitRate;
-        this.peakTimeMultipliers = peakTimeMultipliers;
+        this.peakTimeMultipliers = Collections.unmodifiableMap(new HashMap<>(peakTimeMultipliers));
```

Comment:
```suggestion
        this.peakTimeMultipliers = Map.copyOf(peakTimeMultipliers);
```
Credit goes to IntelliJ for that suggestion...